internal/peer/snapshot/cancelrequest.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package snapshot
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/bccsp"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// cancelRequestCmd returns the cobra command for snapshot cancelrequest command
+func cancelRequestCmd(clientConnection *ClientConnection, cryptoProvider bccsp.BCCSP) *cobra.Command {
+	snapshotCancelRequestCmd := &cobra.Command{
+		Use:   "cancelrequest",
+		Short: "Cancel a snapshot request for the given block number.",
+		Long:  "Cancel a snapshot request for the given block number.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return cancelRequest(cmd, args, clientConnection, cryptoProvider)
+		},
+	}
+
+	flagList := []string{
+		"channelID",
+		"blockNumber",
+		"peerAddress",
+		"tlsRootCertFile",
+	}
+	attachFlags(snapshotCancelRequestCmd, flagList)
+
+	return snapshotCancelRequestCmd
+}
+
+func cancelRequest(cmd *cobra.Command, args []string, cc *ClientConnection, cryptoProvider bccsp.BCCSP) error {
+	if err := validateCancelRequest(); err != nil {
+		return err
+	}
+
+	// Parsing of the command line is done so silence cmd usage
+	cmd.SilenceUsage = true
+
+	// create client connection if not provided
+	if cc == nil {
+		var err error
+		cc, err = NewClientConnection(cmd.Name(), cryptoProvider)
+		if err != nil {
+			return err
+		}
+	}
+
+	request := &pb.SnapshotRequest{
+		ChannelId: channelID,
+		Height:    blockNumber,
+	}
+	signedRequest, err := signSnapshotRequest(cc.Signer, request)
+	if err != nil {
+		return err
+	}
+
+	_, err = cc.SnapshotClient.Cancel(context.Background(), signedRequest)
+	if err != nil {
+		return errors.WithMessage(err, "failed to cancel the request")
+	}
+
+	fmt.Println("Snapshot request cancelled successfully")
+	return nil
+}
+
+func validateCancelRequest() error {
+	if channelID == "" {
+		return errors.New("the required parameter 'channelID' is empty. Rerun the command with -C flag")
```

Comment:
While it's commonly used throughout fabric, the `errors` package is deprecated and should no longer be used in places where `fmt.Errorf` can be used instead. We've been replacing it where we can as we push new changes.

Reply:
Note `errors.New` is still appropriate but via the golang std library `errors` package and not through `github.com/pkg/errors`. `fmt.Errorf` is the appropriate replacement now for `errors.Errorf`. I think there are still many places in the code where we use `github.com/pkg/errors` for wrapped errors. That being said it's easy enough to create a wrapped error with `fmt.Errorf`: https://golang.org/pkg/errors/#pkg-overview

Reply:
Though this error here is much straight forward at the highest level, but in general, the most useful feature we have leveraging from `github.com/pkg/errors` is the inclusion of stack trace. In the `errors` package `Unwrap` is there but the primary intention is to check for a specific error type in the chain and handle (which we hardly do in Fabric). We can certainly leverage `Unwrap` and built-in mechanism of generating stack trace but that requires us to build almost what we have in `github.com/pkg/errors`. I believe that the Go2 seems to have some proposal for printing line numbers of the source of the wrapped chain at the least. Until then, I would not like to loose on error stack traces which have been quite helpful. BTW, I am not sure whether `github.com/pkg/errors` is deprecated? https://github.com/pkg/errors#roadmap simply states that further feature addition is stopped with the upcoming Go2 error proposals which seems reasonable. So, I would be inclined towards keep using this unless you propose to build something similar within fabric (I hope not) until we get stack printing in some form built-in in the `errors`. Thoughts?

Reply:
@btl5037 why is the `github.com/pkg/errors` deprecated?

Reply:
It's not deprecated per se moreso just entering maintenance mode as mentioned in the link above in favor of Go2 errors once they're out. I'm fine with either approach, but it's something worth being aware of once Go2 is available. Given the usefulness of stack trace, continued usage of `github.com/pkg/errors` is likely fine for now.